feature-libs/cart/saved-cart/components/list/saved-cart-list.component.ts

```diff
@@ -82,6 +94,31 @@ export class SavedCartListComponent implements OnInit, OnDestroy {
     event.stopPropagation();
   }
 
+  protected observeAndReloadSavedCartOnContextChange() {
+    if (this.siteContextService) {
+      const contexts: SiteContextType[] = Object.values(SiteContextType);
+
+      if (!contexts.length) {
+        return;
+      }
+
+      this.subscription.add(
+        from(contexts)
+          .pipe(
+            mergeMap((context: SiteContextType) => {
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Would you consider assigning `siteContextService` to a variable within the `if block` to avoid using the `non-null assertion operator` and disabling linting rules? By assigning `const siteContextService = this.siteContextService;` inside of the `if block` and using `siteContextService.getActiveItem` instead of the class member `this.siteContextService`, typescript will know that it is not null.

Reply:
sure